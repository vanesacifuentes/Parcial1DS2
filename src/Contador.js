//Leidy Vanesa Cifuentes Cod 1629458

import React, { Fragment, useState } from 'react'
import PropTypes from 'prop-types'

const Contador = ({valor1, valor2, nombre})=>{

    const [resultado, setResultado] = useState(0)

    return(
        <Fragment>

           <h1>Se√±or(a) {nombre} la suma o resta de {valor1} y {valor2} es: </h1>
           <h1>Valor total: </h1>
           <h2>{resultado}</h2>
           <button onClick={() => setResultado(valor1 + valor2) }> Sumar </button> 
           <button onClick= {() => setResultado(valor1 - valor2)}> Restar </button>
            
            
        </Fragment>
    )

}

Contador.propTypes={
    contador: PropTypes.number.isRequired

}

export default Contador