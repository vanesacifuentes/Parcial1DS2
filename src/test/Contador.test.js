import Contador from '../Contador'
import { shallow } from 'enzyme'



describe('Pruebas sobre el componente contador',()=>{

   test('Probando con enzyme', ()=>{

    

    const wrapper = shallow(<Contador valor1={4} valor2={2} nombre={'Leidy Vanesa'} />)
    expect(wrapper).toMatchSnapshot()

})

    test('Simular click boton Sumar',() => {
    const wrapper = shallow(<Contador valor1={4} valor2={2} nombre={'Leidy Vanesa'}/>)
    const btn1 = wrapper.find('button').at(0).simulate('click')
    const contando = wrapper.find('h2').text().trim()
    expect(contando).toEqual('6')
  
  })


    test('Simular click boton Restar',() => {
    const wrapper = shallow(<Contador valor1={4} valor2={2} nombre={'Leidy Vanesa'}/>)
    const btn1 = wrapper.find('button').at(1).simulate('click')
    const contando = wrapper.find('h2').text().trim()
    expect(contando).toEqual('2')
  
  })
  
})