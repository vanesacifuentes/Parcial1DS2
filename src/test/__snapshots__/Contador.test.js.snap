// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente contador Probando con enzyme 1`] = `
<Fragment>
  <h1>
    Se√±or(a) 
    Leidy Vanesa
     la suma o resta de 
    4
     y 
    2
     es: 
  </h1>
  <h1>
    Valor total: 
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
     Sumar 
  </button>
  <button
    onClick={[Function]}
  >
     Restar 
  </button>
</Fragment>
`;
